src/sentry/models/group.py

```diff
@@ -1223,14 +1223,21 @@ def update_group_open_period(
         return
 
     if new_status == GroupStatus.RESOLVED:
+        if activity is None:
```

Comment:
issue platform is guaranteed to not pass the activity?

Reply:
Issue platform's status updates go through `Group.update_group_status`, which would create the activity and include that in the call to `update_group_open_period`

Reply:
ideally, never. i have another PR up which adds a warning log here but the query is intended to keep the activity linked if the caller ever neglects to include the activity

Reply:
if you make the type `Activity | None` -> `Activity`, does that resolve the issue? 🤔